pandas/tests/frame/methods/test_info.py

```diff
@@ -360,7 +361,7 @@ def test_info_memory_usage():
     df = DataFrame(data)
     df.columns = dtypes
 
-    df_with_object_index = DataFrame({"a": [1]}, index=["foo"])
+    df_with_object_index = DataFrame({"a": [1]}, index=Index(["foo"], dtype=object))
```

Comment:
Similar comment on all of these - might be overlooking something simple but unsure why `dtype=object` is the solution

Reply:
Because what we are testing here is that _if_ you have object dtype, the full memory usage is not known (and you get this "+"):

```
In [1]: df = DataFrame({"a": ["a", "b"]})

In [2]: df.info()
<class 'pandas.DataFrame'>
RangeIndex: 2 entries, 0 to 1
Data columns (total 1 columns):
 #   Column  Non-Null Count  Dtype 
---  ------  --------------  ----- 
 0   a       2 non-null      object
dtypes: object(1)
memory usage: 148.0+ bytes

In [3]: pd.options.future.infer_string = True

In [4]: df = DataFrame({"a": ["a", "b"]})

In [5]: df.info()
<class 'pandas.DataFrame'>
RangeIndex: 2 entries, 0 to 1
Data columns (total 1 columns):
 #   Column  Non-Null Count  Dtype 
---  ------  --------------  ----- 
 0   a       2 non-null      string
dtypes: string(1)
memory usage: 150.0 bytes
```

So `148.0+ bytes` vs `150.0 bytes`.

Of course I could also update the expected result to be accurate instead of an estimate, but that would 1) complicate the test (since we still have to account for both current and future behaviour), and 2) we still need to test the case of object dtype explicitly anyway.

We should probably add a test specifically for string dtype, though, where we can assert that if you have a proper string dtype the memory is now always the full number and not a lower estimate.

Reply:
Cool - yea would be nice to add a test for exact memory representation with the StringDtype + pyarrow, though can be done separately